docs/api.yaml

```diff
@@ -0,0 +1,925 @@
+          description: OK
+        default:
+          $ref: '#/components/responses/error_response'
+  
+          
+  /v1/tenants/{tenant_name}/subscriptions/{sub_name}/status:
+    parameters:
+      - $ref: '#/components/parameters/tenanat_param'
+      - $ref: '#/components/parameters/sub_param'
+    get:
+      description: >-
+        Get the current status of this subscription.
+      tags: 
+        - Subscription
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/consumer_status'
+        default:
+          $ref: '#/components/responses/error_response'
+
+
+  /v1/tenants/{tenant_name}/subscriptions/{sub_name}/dlt:
+    parameters:
+      - $ref: '#/components/parameters/tenanat_param'
+      - $ref: '#/components/parameters/sub_param'
+      - $ref: '#/components/parameters/http_code'
+      - $ref: '#/components/parameters/from_param'
+      - $ref: '#/components/parameters/to_param'
+      - $ref: '#/components/parameters/msg_id_param'
+      - $ref: '#/components/parameters/group_id_param'
+    get:
+      description: >-
+        View or Peek message(s) from the DLT associated with this subscription.
+        Consumer pointer is not moved.
+        TODO::
+        - Add pagination semantics to the API.
+      tags: 
+        - Consume
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/recieved_message'
+        default:
+          $ref: '#/components/responses/error_response'
+          
+    put:
+      description: >-
+        Unsideline messages from DLT for further consumption.
+      tags: 
+        - Consume
+      responses:
+        '200':
+          description: OK
+        default:
+          $ref: '#/components/responses/error_response'
+   
+     
+components:
+  schemas: 
+    tenant:
+      type: object
+      properties:
+        name:
+          type: string
+          example: "finance-project"
+        quota:
+          type: object
+          # TODO:: To be enabled later, based on enforcement static/dynamic. 
+          # properties:
+          #   storage:
+          #     type: integer
+          #   throughput:
+          #     type: integer
+        
+    tenants:
+      type: array
+      items:
+        $ref: '#/components/schemas/tenant'
+
+    property_bag:
+      type: object
+      additionalProperties:
+        type: string
+        
+    retention_policy:
+      type: object
+      properties:
+        retention_hours:
+          type: integer
+          example: 24
+          default: 48
+          maximum: 76
+          
+    capacity_policy:
+      type: object
+      properties:
+        max_throughput:
+          type: integer
```

Comment:
should we make it more human friendly, like 
10Kb, 1024B, 1MB. The units can be assumed to be per sec. and we only ask for string that can be parsed as unit of some bytes.

or we reduce the scale?. just ask KB or B. MB might be too large.

Reply:
merge both retention and capacity, also only throughput.